custom-login/src/App.jsx

```diff
@@ -21,33 +21,29 @@ import Messages from './Messages';
 import Navbar from './Navbar';
 import Profile from './Profile';
 
-function customAuthHandler({ history }) {
+function customAuthHandler() {
```

Comment:
This example can use the history to set the route, but the callback function should be within a component that is a child of Router so that it can access history from the context.  One difference is that history will use pushState. This is pretty minor and does not really affect functionality. The other difference is that the history router can be configured with a basename. In this case, location.origin/login may not be the correct URL. The router could also be configured to use hash, in that case as well location.origin/login will not be correct.  Our sample is not using basename or hash, so it will work out of the box, but either of these changes will break this. I would lean towards using history.push in this case.

Reply:
I actually think it would be nice to create and configure history object explicitly ahead of time. So it is clear how to configure it with these other options. Right now I think history is being created implicitly with default options.

Reply:
I took the code out of our tests.  I didn't demo how to configure the history object as this is an `okta-react` sample and not a `react-router` sample.

Reply:
In our tests the onAuthRequired callback is done like this, using the history: https://github.com/okta/okta-oidc-js/blob/master/packages/okta-react/test/e2e/harness/src/App.js#L32

Reply:
Somehow in the README, the callback function ended up in the global scope rather than bound inside the App component: https://github.com/okta/okta-oidc-js/tree/master/packages/okta-react#example  `this.props.history` doesn't make sense if the function is global

Reply:
sorry, not our tests, from how we run the redirection inside AuthService.

The extra scope and components help demonstrate why I wasn't showing it here: We add confusion to show how to use features that aren't part of our SDK.  Our goal is to show how redirection happens, and they can use their knowledge of whatever their router may be and scopes on top of that.  If we cover too much it's just a mess.

Reply:
One reason I think it would be nice to have the history object created up front is we often get reports about our stuff not working when the router is configured in a certain way (such as using hash). We currently don't have a simple way to reproduce this. Being able to change parameters and run the sample would help us see the issue.

Reply:
I'm not sure how that will help (that only counts if they have the same configuration, right?) but I'll take a stab

Reply:
Our tests "cheat" to make the example more slim, by altering the index.html that CRA provides to stick the Router context on - not something we want advise everyone to do.

To have a scope where the history is available we have to be within a router context, which means we have to break up the App.jsx.  I've done it as two components within the App.jsx file, take a look.